### DISCUSSION TITLE
List of extensions for ASReview LAB

### DISCUSSION THREAD

**Discussion by Rensvandeschoot** [asreview/asreview#1140] [Show and tell]
ASReview has extensive support for (community-maintained) extensions. In this discussion thread, we list all such extensions. Is an extension missing in this list? Leave a message below. Would you like to read more about developing a new extension? Read [Create an Extension](https://asreview.readthedocs.io/en/latest/API/extension_dev.html) in the documentation. For more information about the use of extensions, see [Extensions in ASReview LAB](https://asreview.readthedocs.io/en/latest/extensions/overview_extensions.html). 

## Metrics, analytics, and plots

[ASReview-datatools](https://github.com/asreview/asreview-datatools) can be used for describing basic properties of a dataset (e.g., number of papers, number of inclusions, the amount of missing data and duplicates), converting file formats via the command line, and cleaning your (input) data by removing duplicate records. Maintained by the [ASReview development team]( https://asreview.nl/about).

[ASReview-wordcloud](https://github.com/asreview/asreview-wordcloud) can be used to create a visual impression of the contents of the titles and abstracts in a dataset via a word cloud. Maintained by the [ASReview development team]( https://asreview.nl/about).

[ASReview-insights](https://github.com/asreview/asreview-insights) can be used to extract performance metrics from the `.asreview` file to evaluate the performance of simulation studies. The metrics can be used to create several plots which can be adjusted via the plotting-API.  Maintained by the [ASReview development team]( https://asreview.nl/about).

## Model extensions

[vocab-extractor](https://github.com/asreview/asreview-extension-vocab-extractor) can be used to extract the vocabulary and vector matrices from the `.asreview` file. For TF-IDF it saves the matrix and the vocabulary to pickle and JSON, respectively, and for doc2vec it stores the entire doc2vec model. Maintained by [jteijema](https://github.com/jteijema/).

[multilingual-feature-extractor](https://github.com/asreview/asreview-multilingual-feature-extractor) implements a multilingual feature extractor algorithm to ASReview LAB. This algorithm allows for the usage of textual data in multiple languages: Arabic, Chinese, Dutch, English, French, German, Italian, Korean, Polish, Portuguese, Russian, Spanish, and Turkish. Maintained by [jteijema](https://github.com/jteijema/).

[17-layer-CNN-classifier](https://github.com/JTeijema/asreview-plugin-model-cnn-17-layer) implements a 17-layer deep convolutional neural network (CNN) model for use in ASReview. Maintained by [jteijema](https://github.com/jteijema/).

[model-switcher]( https://github.com/JTeijema/asreview-plugin-model-switcher) implements a model that switches between two models during simulation runtime. It can be helpful to investigate the performance when later stages of data classification require a different model than the first stage. Maintained by [jteijema](https://github.com/jteijema/).

[CNN-HPO]( https://github.com/BartJanBoverhof/asreview-cnn-hpo) implements a model consisting of two separate classifiers for use during simulation mode. The first *X* amount of iterations (default = 500) are run with a Naïve Bayes model. After the switching, a switch to a CNN is made. Immediately at this switching point, after every 150 iterations, hyperparameter optimization is conducted to find the most suitable CNN architecture for the current iteration. Maintained by [BartJanBoverhof](https://github.com/BartJanBoverhof). 

[wide-doc2vec](https://github.com/JTeijema/asreview-plugin-wide-doc2vec) implements doc2vec with a wider vector size (120).

 [asreview-hyperopt](https://github.com/asreview/asreview-hyperopt) can be used to optimize the hyperparameters of the models in ASReview. Maintained by the [ASReview development team]( https://asreview.nl/about).

## Dataset extensions

[ASReview-COVID19](https://github.com/asreview/asreview-covid19) adds the [CORD19 database]( https://pages.semanticscholar.org/coronavirus-research) and the [COVID19 preprints dataset]( https://github.com/nicholasmfraser/covid19_preprints) to ASReview LAB. Note: This extension is deprecated. It still works for version 0.x of ASReview but datasets are no longer updated.

## Misc

[irr](https://github.com/langtonhugh/asreview_irr) computes the inter-rater reliability. It generates a report with the number of abstracts reviewed by each reviewer, the number flagged as relevant, the number left unreviewed, the Kappa statistic, and the number of abstracts for which researchers A and B disagreed (e.g., A said relevant, B said irrelevant). Maintained by [langtonhugh](https://github.com/langtonhugh).

[XREF2CSV-tool](<https://github.com/erikvullings/xref2csv>) written in NodeJS converts XREF-XML files to CSV files that can be imported to ASreview. Maintained by [erikvullings](https://github.com/erikvullings).

**Comment by jteijema** [comment]
[ASReviews' Makita](https://github.com/asreview/asreview-makita) (MAKe IT Automatic) is a workflow generator for simulation studies using the command line interface of ASReview LAB. Makita can be used to simplify your own research by enabling you to effortlessly generate the framework and code for your simulation study.

**Comment by Rensvandeschoot** [comment]
The [ASReview Notes Export Extension](https://github.com/rohitgarud/asreview-notes-export#asreview-notes-export-extension) is an extension that can be used to export notes from the ASReview GUI while exporting the labeled or partially labeled dataset as .csv file. It solves discussion https://github.com/asreview/asreview/discussions/1220

**Reply by rohitgarud** [reply]
The extension is no longer needed after the latest release of ASReview 1.2 as it is now a core functionality. Look for exported_notes_# in the exported dataset.

**Comment by rohitgarud** [comment]
[Asreview-Ensemble-Classifiers](https://github.com/rohitgarud/Asreview-Ensemble-Classifiers) extension adds a new set of classifiers by ensembling different basic classifiers such as Naive Bayes (NB), Logistic Regression (LR) and Random Forest (RF). A simulation study performed using these classifiers is presented at https://github.com/asreview/asreview/discussions/1334

**Comment by rohitgarud** [comment]
[Asreview-SimilarityClassifier](https://github.com/rohitgarud/Asreview-SimilarityClassifier) extension adds a set of classifiers based on the similarity between the features of the relevant records and the unlabelled records. A simulation study performed using the similarity classifier is presented at https://github.com/asreview/asreview/discussions/1371

**Comment by george-gca** [comment]
[AI Conference Papers](https://github.com/george-gca/ai_conferences_info/tree/main/) is a set of unlabeled data (it doesn't have any paper flagged as useful/not useful) regarding some of the most influential conferences in the AI scene over the last years.

It contains readily importable tsv files containing (title, abstract, url, authors) for the following conferences: AAAI, ACL, COLING, CVPR, EACL, ECCV, EMNLP, Findings of the Association for Computational Linguistics, ICCV, ICLR, ICML, IJCAI, IJCNLP, KDD, NAACL, NeurIPS, SIGCHI, SIGDIAL, SIGGRAPH, TACL, and WACV.

**Comment by george-gca** [comment]
[ASReview Top2Vec](https://github.com/george-gca/asreview-top2vec) is based on the [ASReview Semantic Clustering](https://github.com/asreview/semantic-clusters), but uses [Top2Vec](https://github.com/ddangelov/Top2Vec) underneath. It automatically detects the number of topics it can divide the documents and group them. It can also provide more information about the topics, like most relevant words for each topic. I ditched the visualization inside for one using [Tensorflow's Embedding Projector](https://projector.tensorflow.org/), since it is more flexible, and allows visualizing the documents and its metadata, also changing the dimensionality reduction algorithm to use.

**Reply by jteijema** [reply]
Cool! I'll definitely check this out.

**Comment by mattebso** [comment]
**living-review-updater** ([GitHub](https://github.com/mattebso/living-review-updater) · [PyPI](https://pypi.org/project/living-review-updater/)) adds an `asreview livingreview` subcommand for keeping a finished review up to date. It fetches new records from PubMed and Europe PMC since your last search, deduplicates them against everything you've already screened, ranks them using the prior decisions read directly from your `.asreview` project file, and writes a digest plus a ranked RIS file ready for a new screening round in ASReview.

It ranks and flags only — it never auto-excludes. In a retrospective backtest on 26 SYNERGY reviews (true original→update time split), the median screening-effort saving was ~71% at 95% recall, with a wide per-review spread, which is why it calibrates on each team's own decisions and refuses to act as a stopping rule on thin data. Details and evaluation scripts are in the repo's `eval/`.

Install: `pip install living-review-updater[asreview]` (MIT).

Disclosure: this tool is developed by an AI agent (Claude), with a human collaborator handling review and publishing. Issues and feedback very welcome.
